Add: [Win32] regression testing with MSVC

glx22 opened the pull request.

glx22 committed:
Fix: [Win32] don't mess with std[out|err|in] if we can't allocate a console

glx22 committed:
Add: [Win32] regression test script

glx22 committed:
Add: [Win32] regression project files

glx22 commented:
If the CI tries the updated version, it should now fail at regression because OpenGFX is probably not present

TrueBrain commented:
I canceled the build, as without gfx, it just "hangs" for ever. Which is rather annoying.

Guess you need something like this:

`  - bash: |
      set -ex
      mkdir -p ~/Documents/OpenTTD/baseset
      cd ~/Documents/OpenTTD/baseset
      curl -L https://binaries.openttd.org/extra/opengfx/0.5.2/opengfx-0.5.2-all.zip > opengfx-0.5.2-all.zip
      unzip opengfx-0.5.2-all.zip
      rm -f opengfx-0.5.2-all.zip
    displayName: 'Install OpenGFX'`

Before `Build` step. Feel free to modify it, or make it non-bash (that would be optimal :D). I have no clue if the destination is correct etc. Never got to test it ;)

Additionally, a question: if it is just a vbs file to run, is it needed to have different configuration for the regression? Can't we just have the Debug/Release, and that we run the vbs script after that to run regression? (honest question)

glx22 commented:
I think it hangs because a message box opens when the gfx is not found.

There are extra steps around the vbs file run, that's why I added a projects in the solution. And the extra configuration were to run regression from MSBuild, but indeed it's possible to do it from outside (the regression project just calls a batch script).

The main question is where to put openGFX :). Looking in README there are many options, and I don't know what's best for azure pipelines. I think the easy way is to put it directly in bin/baseset, but I'm not sure.

For now I removed the added configurations as they were somehow overkill. And I'm working on a new yml version.

glx22 committed:
Add: [AzurePipelines] run regression with MSVC

LordAro commented on azure-pipelines-ci.yml:
```diff
@@ -34,14 +34,24 @@ jobs:
+      unzip opengfx-0.5.2-all.zip
+      rm -f opengfx-0.5.2-all.zip
+    displayName: 'Install OpenGFX'
+    workingDirectory: '$(Build.SourcesDirectory)/bin/baseset'
```
Depending on how releases eventually get done, this might cause issues/complications with packaging?

glx22 replied:
install.nsi doesn't pick tar files so it should be ok

LordAro commented on src/os/windows/win32.cpp:
```diff
@@ -300,7 +300,7 @@ void CreateConsole()
 	_has_console = true;
 
-	AllocConsole();
+	if (!AllocConsole()) return;
```
maybe a comment?

glx22 commented:
Ok it fails, but it's a progress (no hang this time). Looking at the logs my install GFX task doesn't work, and the batch fails because it's not run in the VS environment

TrueBrain commented on azure-pipelines-ci.yml:
```diff
@@ -34,14 +34,25 @@ jobs:
       maximumCpuCount: true
-  # Running the regression is currently not possibe via MSVC (console is not redirected)
+  - script: |
+      call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x86
```
is x86 also working for 64bit?

glx22 replied:
Yes, it's just to get access to editbin.exe, and the platform doesn't matter here. Would be different if we needed to use compiler or linker.
